Firestore/core/src/firebase/firestore/immutable/llrb_node.h

```diff
@@ -258,6 +288,63 @@ LlrbNode<K, V> LlrbNode<K, V>::InnerInsert(const K& key,
   return result;
 }
 
+template <typename K, typename V>
+template <typename Comparator>
+LlrbNode<K, V> LlrbNode<K, V>::erase(const K& key,
+                                     const Comparator& comparator) const {
+  LlrbNode root = InnerErase(key, comparator);
+  root.FixRootColor();
+  return root;
+}
+
+template <typename K, typename V>
+template <typename Comparator>
+LlrbNode<K, V> LlrbNode<K, V>::InnerErase(const K& key,
+                                          const Comparator& comparator) const {
+  if (empty()) {
+    // Empty node already frozen
+    return LlrbNode{};
+  }
+
+  LlrbNode n = Clone();
+
+  bool descending = comparator(key, n.key());
+  if (descending) {
+    if (!n.left().empty() && !n.left().red() && !n.left().left().red()) {
```

Comment:
Optional: what exactly does this check do (I see it's repeated in `RemoveMin`, so it must represent some interesting property of the tree)? Would it be possible to create a named variable (or a helper function) to make the meaning of this configuration clearer? I presume it avoids having a chain of too many black nodes?